version: "3.3"

services:
    ms-user:
        build: .
        # ports: 
        #     - "7000:7000"
        # container_name: ms-user
        environment:
            - DB_HOST=db
            - DB_PORT=3306
            - DB_DATABASE=authme
            - DB_USER=root
            - DB_PASSWORD=mariposa
            - JWT_SECRET_KEY = 'clave-secreta-de-la-app'
            - JWT_ACCESS_TOKEN_EXPIRES = 3600

        labels:
            - "traefik.enable=true"
            - "traefik.http.routers.ms-user.rule=Host(`ms-otp.authme.localhost`) && Path(`/user`)"
            - "traefik.http.routers.ms-user.tls=true"
            - "traefik.http.services.ms-user.loadbalancer.server.port=5000"
            - "traefik.http.middlewares.ms-user-circuitbreaker.circuitbreaker.expression=NetworkErrorRatio() > 0.5"
            - "traefik.http.routers.ms-user.middlewares=ms-user-circuitbreaker"
            - "traefik.docker.network=red"
        networks:
            - traefik-network
            
        volumes:
            - ./main:/app/main
            - ./app.py:/app/app.py


networks:
    traefik-network:
        external: true